src/modules/esl-base-element/core/esl-base-element.ts

```diff
@@ -37,18 +38,18 @@ export abstract class ESLBaseElement extends HTMLElement {
   }
 
   /** Subscribes `handler` method marked with `@listen` decorator */
-  public $$on(handler: ESLListenerHandler): void;
+  public $$on(handler: ESLListenerHandler): ESLEventListener[];
   /** Subscribes `handler` function by the passed DOM event descriptor {@link ESLListenerDescriptor} or event name */
   public $$on<EType extends keyof ESLListenerEventMap>(
-    handler: ESLListenerHandler<ESLListenerEventMap[EType]>,
-    event: EType | ESLListenerDescriptor<EType>
-  ): void;
-  public $$on(handler: ESLListenerHandler, descriptor?: keyof ESLListenerEventMap | ESLListenerDescriptor): void {
-    return EventUtils.subscribe(this, handler, descriptor);
+    event: EType | ESLListenerDescriptor<EType>,
+    handler: ESLListenerHandler<ESLListenerEventMap[EType]>
+  ): ESLEventListener[];
+  public $$on(event: any, handler?: any): ESLEventListener[] {
```

Comment:
I'm not sure if type of event and handler here should be any

Reply:
It's not visible for outside user as it's function definition with overloads, so TS will provide only overloads signatures to the outside world.
Regarding types inside of the function, it was just easy for me say TS that "here I'm care about types by overself" instead of overcomlicate it just because this method is only a proxy for the `EventUtils.subscribe`